ted2zim/scraper.py

```diff
@@ -258,471 +308,217 @@ def dump_data(self):
+                save_large_file(video_thumbnail, thumbnail_path)
             else:
-                print(
-                    "thumbnail.jpg already exist. Skipping video "
-                    + video_title.encode("utf-8")
-                )
+                logger.debug(f"Thumbnail already exists for {video_title}")
+
+            # recompress if necessary
+            post_process_video(
+                video_dir, video_id, self.video_format, self.low_quality,
+            )
 
     def download_subtitles(self):
-        """
-        Download the subtitle files, generate a WebVTT file
-        and save the subtitles in
-        TED/build/{video id}/subs_{language code}.vtt.
-        """
-        self.load_metadata()
+        # Download the subtitle files, generate a WebVTT file
+        # and save the subtitles in
+        # build_dir/{video id}/subs/subs_{language code}.vtt
+        self.load_meta_from_file()
         for video in self.videos:
             video_id = str(video[0]["id"])
             video_title = video[0]["title"]
             video_subtitles = video[0]["subtitles"]
-            subs_dir = path.join(self.scraper_dir, video_id, "subs")
-
-            if not path.exists(subs_dir):
-                os.makedirs(subs_dir)
+            video_dir = self.build_dir.joinpath(video_id)
+            subs_dir = video_dir.joinpath("subs")
+            if not subs_dir.exists():
+                subs_dir.mkdir(parents=True)
             else:
-                print("Subtitles already exist. Skipping video.")
+                logger.debug(f"Subs dir exists already")
                 continue
 
             # download subtitles
-            print("Downloading subtitles... " + video_title.encode("utf-8"))
+            logger.debug(f"Downloading subtitles for {video_title}")
             for subtitle in video_subtitles:
                 sleep(0.5)
                 subtitle_file = WebVTTcreator(subtitle["link"], 11820).get_content()
-                if subtitle_file == False:
+                if not subtitle_file:
                     video[0]["subtitles"].remove(subtitle)
-                    pass
-                subtitle_file = subtitle_file.encode("utf-8")
-                subtitle_file_name = "subs_{}.vtt".format(subtitle["languageCode"])
-                subtitle_file_name = path.join(subs_dir, subtitle_file_name)
-                with open(subtitle_file_name, "w") as sub_file:
+                subtitle_file_name = subs_dir.joinpath(
+                    f"subs_{subtitle['languageCode']}.vtt"
+                )
+                with open(subtitle_file_name, "w", encoding="utf-8") as sub_file:
                     sub_file.write(subtitle_file)
-        self.dump_data()  # To save info than some videos have, finaly not subtitle
 
-    def load_metadata(self):
-        """
-        Load the dumped json meta-data file.
-        """
+        # save the info that some videos don't have subtitlescategories
```

Comment:
?

Reply:
Okay. So we save the data if subtitle files could not be created so that the HTML rendering takes this into consideration

Reply:
Though I missed a continue here. Also added this incident to logs.